docs/docs/reference/dropped/weak-conformance.md

```diff
@@ -0,0 +1,78 @@
+---
+layout: doc-page
+title: Dropped: Weak Conformance
+---
+
+In some situations, Scala used a {\em weak conformance} relation when
+testing type compatibility or computing the least upper bound of a set
+of types.  The principal motivation behind weak conformance was to
+make as expression like this have type `List[Double]`:
+
+    List(1.0, math.sqrt(3.0), 0, -3.3) // : List[Double]
+
+It's "obvious" that this should be a `List[Double]`. However, without
+some special provision, the least upper bound of the lists's element
+types `(Double, Double, Int, Double)` would be `AnyVal`, hence the list
+expression would be given type `List[AnyVal]`.
+
+A less obvious example is the following one, which was also typed as a
+`List[Double]`, using the weak conformance relation.
+
+    val n: Int = 3
+    val c: Char = 'X'
+    val n: Double = math.sqrt(3.0)
+    List(n, c, d) // used to be: List[Double], now: List[AnyVal]
+
+Here, it is less clear why the type should be widened to
+`List[Double]`, a `List[AnyVal` seems to be an equally valid -- and
+more principled -- choice.
+
+To simplify the underlying type theory, Dotty drops the notion of weak
+conformance altogether. Instead, it provides more flexibility when
+assigning a type to a constant expression. The rule is:
+
+ - If a list of expressions `Es` appears as one of
+
+     - the elements of a vararg parameter, or
+     - the alternatives of an if-then-else or match expression, or
+     - the body and catch results of a try expression,
+
+   and all expressions have primitive numeric types, but they do not
+   all have the same type, then the following is attempted: Every
+   constant expression `E` in `Es` is widened to the least primitive
+   numeric value type above the types of all expressions in `Es`. Here
+   _above_ and _least_ are interpreted according to the ordering given
+   below.
+
+
+                  Double
+                 /      \
+               Long    Float
+                 \     /
+                   Int
+                 /    \
+              Short   Char
+                |
+              Byte
+
+    If these widenings lead to all expressions `Es` having the same type,
+    we use the transformed list of expressions instead of `Es`, otherwise
+    we use `Es` unchanged.
+
+__Examples:__
+
+    inline val b: Byte = 3
```

Comment:
Would be good if there was some explanation of how the rules differ between `val` and `inline val`.